spec/models/course_enrichment_spec.rb

```diff
@@ -91,36 +91,75 @@
+    end
+    subject { course_enrichment }
 
     context 'fee based course' do
-      let(:course_enrichment_traits) { [:with_fee_based_course] }
-
       it { should_not validate_presence_of(:salary_details).on(:publish) }
       it { should validate_presence_of(:fee_uk_eu).on(:publish) }
-
-      context 'valid content' do
-        it { should be_valid :publish }
-      end
-
-      context 'invalid content' do
-        let(:course_enrichment_traits) { %i[with_fee_based_course with_invalid_content] }
-        it { should_not be_valid :publish }
+      it { should validate_presence_of(:about_course).on(:publish) }
+      it { should validate_presence_of(:qualifications).on(:publish) }
+
+
+      context 'invalid content exceed word count fields' do
+        let(:course_enrichment) do
+          create(:course_enrichment, :with_fee_based_course,
+                 about_course:  Faker::Lorem.sentence(400 + 1),
+                 interview_process:  Faker::Lorem.sentence(250 + 1),
+                 qualifications:  Faker::Lorem.sentence(100 + 1),
+                 how_school_placements_work:  Faker::Lorem.sentence(350 + 1),
+                 fee_details:  Faker::Lorem.sentence(250 + 1),
+                 salary_details:  Faker::Lorem.sentence(250 + 1))
+        end
+
+        subject do
+          course_enrichment.valid? :publish
+          course_enrichment.errors
+        end
+
+        its([:about_course]) { should match_array ['^Reduce the word count for about course'] }
+        its([:interview_process]) { should match_array ['^Reduce the word count for interview process'] }
+        its([:qualifications]) { should match_array ['^Reduce the word count for qualifications'] }
+        its([:how_school_placements_work]) { should match_array ['^Reduce the word count for how school placements work'] }
+        its([:fee_details]) { should match_array ['^Reduce the word count for fee details'] }
+        its([:salary_details]) { should be_empty }
       end
     end
 
     context 'salary based course' do
-      let(:course_enrichment_traits) { [:with_salary_based_course] }
+      let(:course_enrichment) do
+        create(:course_enrichment, :with_salary_based_course)
+      end
 
       it { should validate_presence_of(:salary_details).on(:publish) }
+      it { should validate_presence_of(:about_course).on(:publish) }
+      it { should validate_presence_of(:qualifications).on(:publish) }
       it { should_not validate_presence_of(:fee_uk_eu).on(:publish) }
-      context 'valid content' do
-        it { should be_valid :publish }
-      end
 
-      context 'invalid content' do
-        let(:course_enrichment_traits) { %i[with_salary_based_course with_invalid_content] }
-        it { should_not be_valid :publish }
+      context 'invalid content exceed word count fields' do
+        let(:course_enrichment) do
+          create(:course_enrichment, :with_salary_based_course,
+                 about_course:  Faker::Lorem.sentence(400 + 1),
```

Comment:
Same here, haven't we tested this already?

Reply:
Kinda, there is two types of enrichment for a given course, a course that is fee based or a course that is salary based.

The validation falls in to 3 categories, common to both and specifically to only one type (ie salary details vs fee details). Validation is only triggered when we try to publish it.

Reply:
Refactored 

As given a enrichment with a x based course

Test common
Test specify to x based course
Test the other to y based course